### PR TITLE
Single node async 

### PR DESCRIPTION
**Pull request by vthacker**
Make query code for a single node async. We need this for multi-node search since everything needs to be non-blocking from botton up

Attempt 2 where original PR was https://github.com/slackhq/kaldb/pull/52

### CODE REVIEW

File: src/test/java/com/slack/kaldb/chunk/ChunkManagerTest.java [hunk 1]
```diff
@@ -364,10 +371,98 @@ public void testAddMessagesToChunkWithRollover() throws IOException {
...
+    testChunkManagerSearch(chunkManager, "Message1", 0, 3, 0, 0, MAX_TIME);
+    testChunkManagerSearch(chunkManager, "Message11", 0, 3, 0, 0, MAX_TIME);
+    testChunkManagerSearch(chunkManager, "Message21", 0, 3, 0, 0, MAX_TIME);
+
+    chunkManager
+        .getChunkMap()
+        .values()
+        .forEach(chunk -> chunk.setLogSearcher(new IllegalArgumentLogIndexSearcherImpl()));
+
+    try {
```
Comment (mansu) [inline src/test/java/com/slack/kaldb/chunk/ChunkManagerTest.java:460]: Please use a catchThrowable here.

File: src/main/java/com/slack/kaldb/chunk/ChunkManager.java [hunk 2]
```diff
@@ -263,18 +272,42 @@ public void rollOverActiveChunk() {
...
+            .stream()
+            .filter(
+                chunk ->
+                    chunk.containsDataInTimeRange(
+                        query.startTimeEpochMs / 1000, query.endTimeEpochMs / 1000))
+            .map(
+                (chunk) ->
+                    // TODO: make 10 configurable via SearchRequest
+                    CompletableFuture.supplyAsync(() -> chunk.query(query), queryExecutorService)
+                        .completeOnTimeout(empty, 10, TimeUnit.SECONDS))
```
[the remark this answers is not in the document]
Comment (vthacker) [reply src/main/java/com/slack/kaldb/chunk/ChunkManager.java]: done!

File: src/test/java/com/slack/kaldb/server/KaldbLocalSearcherTest.java [hunk 3]
```diff
@@ -299,15 +304,26 @@ public void testKalDbBadArgSearch() throws IOException {
...
+        kaldbLocalSearcher.doSearch(
+            searchRequestBuilder
+                .setIndexName(MessageUtil.TEST_INDEX_NAME)
+                .setQueryString("Message1")
+                .setStartTimeEpochMs(chunk1StartTimeMs)
+                .setEndTimeEpochMs(chunk1EndTimeMs)
+                .setHowMany(0)
+                .setBucketCount(0)
+                .build());
+    try {
```
[the remark this answers is not in the document]
Comment (vthacker) [reply src/test/java/com/slack/kaldb/server/KaldbLocalSearcherTest.java]: done!

File: src/test/java/com/slack/kaldb/chunk/ChunkManagerTest.java (hunk 1 above)
Comment (vthacker) [reply src/test/java/com/slack/kaldb/chunk/ChunkManagerTest.java]: done!

File: src/main/java/com/slack/kaldb/chunk/ChunkManager.java [hunk 4]
```diff
@@ -74,6 +73,12 @@
   private final long rolloverFutureTimeoutMs;
   private ListenableFuture<Boolean> rolloverFuture;
 
+  // TODO: We want to move this to the config eventually
+  public static final int QUERY_TIMEOUT = 10;
```
[the remark this answers is not in the document]
Comment (vthacker) [reply src/main/java/com/slack/kaldb/chunk/ChunkManager.java]: done!

File: src/test/java/com/slack/kaldb/logstore/search/SearchResultAggregatorImplTest.java [hunk 5]
```diff
@@ -405,4 +422,13 @@ public void testSimpleSearchResultsAggIgnoreHitsSafely() {
       assertThat(b.getCount() == 10 || b.getCount() == 0).isTrue();
     }
   }
+
+  public CompletableFuture<List<SearchResult<LogMessage>>> makeAsncResults(
```
Comment (mansu) [inline src/test/java/com/slack/kaldb/logstore/search/SearchResultAggregatorImplTest.java:426]: typo missing y in async.

Comment (vthacker) [reply src/test/java/com/slack/kaldb/logstore/search/SearchResultAggregatorImplTest.java:426]: haha! good catch
fixing

**Review by mansu — APPROVED** [review]

**Pull request merged**